jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -126,26 +207,53 @@ public HttpCookie(String name, String value, String domain, String path, long ma
         _comment = comment;
         _version = version;
         _expiration = maxAge < 0 ? -1 : NanoTime.now() + TimeUnit.SECONDS.toNanos(maxAge);
-        _attributes = (attributes == null ? Collections.emptyMap() : attributes);
+        Map<String, String> attrs = null;
+        if (attributes == null)
+            attrs = Collections.emptyMap(); // unmodifiable empty map
+        else
+        {
+            attrs = new HashMap<>(attributes); // create new map, to only capture relevant attributes
```

Comment:
The API Cookie class uses a tree map that is case insensitive: `new TreeMap<>(String.CASE_INSENSITIVE_ORDER);`. 

We should probably mimic.